models/forum.server.model.ts

```diff
@@ -0,0 +1,163 @@
+import Forum from "../config/db_schemas/forum.schema";
+import Comment from "../config/db_schemas/comment.schema";
+
+/**
+ * Insert a new forum post to the database
+ * @param params object containing forum post attributes
+ * @param done function callback, returns status code, and message if error, or JSON if successful
+ */
+export function insertPost (params, done) {
+    // Set forum post attributes
+    const
+      userID = params.userID,
+      communityID = params.communityID,
+      title = params.title,
+      bodyText = params.bodyText,
+      edited = false,
+      upVotes = 0,
+      downVotes = 0,
+      attachments = params.attachments,
+      comments = ['']
+
+    // Create new forum post document
+    const newPost = new Forum({
+        userID,
+        communityID,
+        title,
+        bodyText,
+        edited,
+        upVotes,
+        downVotes,
+        attachments,
+        comments
+    })
+
+    // Save new forum post document to database collection
+    newPost.save()
+      .then((res) => {
+          return done(res)
+      })
+      .catch((err) => {
+          // Forum post is already in the database with unique attributes, return duplicate conflict error
+          if (err.code === 11000) {
```

Comment:
this code seems like magic to me

Reply:
it's a mongodb error code. I think we should fix this in another ticket later